staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -88,6 +103,42 @@ var (
 		},
 		[]string{priorityLevel, flowSchema},
 	)
+	windowedRequestCountsMin = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "windowed_request_count_min",
+			Help:      "Minimum number of requests queued, executing in previous window",
+		},
+		[]string{priorityLevel, phase},
+	)
+	windowedRequestCountsMax = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "windowed_request_count_max",
+			Help:      "Maximum number of requests queued, executing in previous window",
+		},
+		[]string{priorityLevel, phase},
+	)
+	windowedRequestCountsAvg = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "windowed_request_count_avg",
```

Comment:
Is the average time-weighted?  I.e. if my requests queued is 0 for most of the window, and then 10 at the very end, do I get 5 here, or close to 0?

Reply:
It is time-weighted.